### PR TITLE
SafeHandle and IDisposable pattern implementation

### PR DESCRIPTION
**Pull request by JeroMiya**
This is the implementation of this issue:
https://github.com/OSVR/Managed-OSVR/issues/9

### CODE REVIEW

File: ClientKit/ClientKit.cs [hunk 1]
```diff
@@ -19,6 +19,8 @@
 ﻿using System;
 using System.Runtime.InteropServices;
 using System.Text;
+using Microsoft.Win32.SafeHandles;
```
Comment (rpavlik) [inline ClientKit/ClientKit.cs:22]: I'm assuming that despite this name it's available in Mono cross-platform?

File: ClientKit/ClientKit.cs [hunk 2]
```diff
@@ -30,6 +32,18 @@ namespace OSVR
...
     namespace ClientKit
     {
+        public sealed class SafeClientContextHandle : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            public SafeClientContextHandle() : base(true) { }
+
+            [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+            protected override bool ReleaseHandle()
+            {
+                System.Diagnostics.Debug.WriteLine("Client shutdown");
```
Comment (rpavlik) [inline ClientKit/ClientKit.cs:42]: Do we need to preface this with something like `[OSVR]` or does .NET does something like that automagically?

**Comment by rpavlik** [comment]
Looks good on visual inspection.

File: ClientKit/ClientKit.cs (hunk 1 above)
Comment (JeroMiya) [inline ClientKit/ClientKit.cs:22]: That's correct.

File: ClientKit/ClientKit.cs (hunk 2 above)
Comment (JeroMiya) [inline ClientKit/ClientKit.cs:42]: Hmm, no automagic that I know of. I'll add "[OSVR]"  in front of these messages. Also, should this say "[OSVR] ClientContext shutdown" or just "[OSVR] Client shutdown"?

Comment (rpavlik) [inline ClientKit/ClientKit.cs:42]: Probably ClientContext.

**Pull request merged**